libs/src/api/file.js

```diff
@@ -1,14 +1,49 @@
 const { Base, Services } = require('./base')
 const axios = require('axios')
+const Utils = require('../utils')
 
-const DEFAULT_TIMEOUT = 250
+const CancelToken = axios.CancelToken
 
 // Public gateways to send requests to, ordered by precidence.
 const publicGateways = [
   'https://ipfs.io/ipfs/',
   'https://cloudflare-ipfs.com/ipfs/'
 ]
 
+// Races requests for file content
+async function raceRequests (
+  urls,
+  callback
+) {
+  const sources = []
+  const requests = urls.map(async url => {
+    const source = CancelToken.source()
+    sources.push(source)
+
+    return axios({
+      method: 'get',
+      url,
+      responseType: 'blob',
+      cancelToken: source.token
+    })
+      .then(response => ({
+        blob: response,
+        url
+      }))
+      .catch((thrown) => {
```

Comment:
maybe want to log if there is a non-axios cancel err?

Reply:
gonna punt on this because it'll be really spammy (e.g. we're going to try many creator nodes that don't have the content). We should add a specific logging event for these cases though and catch it in fullstory or heap or something